### PR TITLE
feature/Input alert

### PR DESCRIPTION
**Pull request by davidfranquet**
#### 📲 What

- Created a new input alert to handle the rename crop/username functionality
- Integrated it on rename crop & rename profile

#### 🤔 Why
		
We need this input alert to write the desired value when renaming a crop or a username
		
#### 👀 See
		
ezgif com-video-to-gif-9

		 
#### ✅ Acceptance criteria

- [ ] Design Review for UI with BA completed. 
- [ ] Manually tested and verified on Android device/emulator (min. Lollipop 5.1)
- [ ] Showcase video / automation test
- [ ] Passing all tests
- [ ] Rebased/merged with latest changes from development and re-tested
- [ ] Removed unsed comments. TODOs must have JIRA for future work.

#### Coding Standards Checklist
- [ ] unit tests 80% coverage on testable code (functions, methods, class)
- [ ] (Architecture) redux state, reducers, actions, middleware defined under lib/redux
- [ ] ui elements defined under lib/ui. 
- [ ] Strings ready for localisation (e.g. defined in lib/utils/strings.dart)
- [ ] Assets (images/icons path) defined in lib/utils/assets.dart
- [ ] Colors defined in lib/utils/colors.dart
- [ ] Dimensions (ie margins padding) defined in lib/utils/dimens.dart
- [ ] TextStyles defined in lib/utils/styles.dart

Recommended Style Guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo

#### 🕵️‍♂️ How to test

Notes for QA

#### Reviewers

@farmsmart/amido @farmsmart/wamf

### CODE REVIEW

**Review by alexanishchenko88 — DISMISSED** [review]

File: lib/ui/common/InputAlert.dart [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+            borderRadius: _Constants.actionCornerRadius,
+          ),
+        ),
+      ),
+    );
+    return listBuilder;
+  }
+
+  confirmAndDismiss(BuildContext context) {
+    _viewModel.confirmInputAction(_textFieldController.text);
```
Comment (LeeMatthewHiggins) [inline lib/ui/common/InputAlert.dart]: wonder if we should allow null action here, and if the cancel name not set it becomes a OK dialog message?

File: lib/ui/profile/Profile.dart [hunk 2]
```diff
@@ -604,18 +609,42 @@ class Profile extends StatelessWidget {
...
+
+  InputAlert _renameInputAlert(ProfileViewModel viewModel) {
+    return InputAlert(
+      viewModel: InputAlertViewModel(
+          cancelActionText: _LocalisedStrings.cancel(),
+          confirmActionText: _LocalisedStrings.confirm(),
+          titleText: _LocalisedStrings.renameProfile(),
+          hint: _LocalisedStrings.username(),
+          confirmInputAction: (value) {
+            //TODO Add the viewmodel rename func here
```
Comment (LeeMatthewHiggins) [inline lib/ui/profile/Profile.dart]: let´s add the interface here and have the TODO in the profile repo? then we don't have to come back the this widget.. so add a rename action too view model, and in the block call a new rename function on the profile repo, and then just have a todo in there.

**Review by LeeMatthewHiggins — CHANGES_REQUESTED** [review]

File: lib/ui/common/InputAlert.dart (hunk 1 above)
Comment (davidfranquet) [reply lib/ui/common/InputAlert.dart]: Cancel is mandatory, so cancel text should be always informed

Comment (davidfranquet) [reply lib/ui/common/InputAlert.dart]: DONE

File: lib/ui/profile/Profile.dart (hunk 2 above)
Comment (davidfranquet) [reply lib/ui/profile/Profile.dart]: DONE

**Review by LeeMatthewHiggins — COMMENTED** [review]
Could you check this works on small screens, worried the keyboard will cover the input..

**Comment by davidfranquet** [comment]
> Could you check this works on small screens, worried the keyboard will cover the input..

Good one! Is not covering the input, but is covering the buttons. I've fixed that.
Screenshot_20190909-095847

**Review by LeeMatthewHiggins — DISMISSED** [review]
LGTM 👍

File: lib/model/bloc/profile/ProfileDetailProvider.dart [hunk 3]
```diff
@@ -163,4 +164,8 @@ class ProfileDetailProvider
     //TODO Update image for User
     print(savedImagePath);
   }
+
+  void _renameProfile(String username) {
+    //TODO Missing repo functionality update profile username
```
Comment (albertmiro) [inline lib/model/bloc/profile/ProfileDetailProvider.dart:169]: That would be done in another PR?

File: lib/ui/common/InputAlert.dart [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+            decoration: BoxDecoration(
+              borderRadius: _Constants.cornerRadius,
+              color: _style.backgroundColor,
+            ),
+            child: Padding(
+              padding: _Constants.alertInnerPadding,
+              child: Column(
+                crossAxisAlignment: CrossAxisAlignment.start,
+                mainAxisSize: MainAxisSize.min,
+                children: <Widget>[
```
Comment (albertmiro) [inline lib/ui/common/InputAlert.dart:148]: Can we decompose that in different methods?

File: lib/ui/myplot/PlotDetail.dart [hunk 5]
```diff
@@ -203,7 +211,21 @@ class _PlotDetailState extends State<PlotDetail> {
...
+    InputAlert.present(_renameInputAlert(viewModel), context);
+  }
+
+  InputAlert _renameInputAlert(PlotDetailViewModel viewModel) {
+    return InputAlert(
+      viewModel: InputAlertViewModel(
+        cancelActionText: _LocalisedStrings.cancelAction(),
+        confirmActionText: _LocalisedStrings.confirm(),
+        titleText: _LocalisedStrings.renameAction(),
+        hint: _LocalisedStrings.cropName(),
```
Comment (albertmiro) [inline lib/ui/myplot/PlotDetail.dart:223]: Not sure if I would name this string as `cropName` or `hintCropName` , what do you think?

**Review by LeeMatthewHiggins — APPROVED** [review]

File: lib/model/bloc/profile/ProfileDetailProvider.dart (hunk 3 above)
Comment (davidfranquet) [reply lib/model/bloc/profile/ProfileDetailProvider.dart:169]: yep

File: lib/ui/myplot/PlotDetail.dart (hunk 5 above)
Comment (davidfranquet) [reply lib/ui/myplot/PlotDetail.dart:223]: cropName is fine, it's a LocalisedString and we name it for what it is. It is "Crop name" not "Hint crop name" 😄

**Review by albertmiro — APPROVED** [review]
